### ISSUE TITLE
Bitcode serialization fails with unset Python_EXECUTABLE variable.

### ISSUE THREAD

**Issue by tgrant-nv** [AcademySoftwareFoundation/OpenShadingLanguage#1503]
### Problem

The custom command to serialize the bitcode for liboslexec fails if the `Python_EXECUTABLE` CMake variable is not set. This is due to a slight change in the invocation of the serialization script:

```diff
-        COMMAND python "${CMAKE_SOURCE_DIR}/src/build-scripts/serialize-bc.py" ${llvm_bc} ${llvm_bc_cpp} ${prefix}
+        COMMAND ${Python_EXECUTABLE} "${CMAKE_SOURCE_DIR}/src/build-scripts/serialize-bc.py" ${llvm_bc} ${llvm_bc_cpp} ${prefix}
```

This can be worked around by setting `Python_EXECUTABLE` or setting `USE_PYTHON=ON` at configure time. 

Perhaps for the sake of compatibility it would be better to either set a default value so that the build will work as before, or to issue a warning or error if the variable is not set?

### Versions

* OSL branch/version:  main (033dae37)
* OS: Ubuntu 20.04
* C++ compiler:  clang 12.0.0
* LLVM version:  12.0.0
* OIIO version: N/A

**Comment by fpsunflower** [comment]
Ah apologies, I thought python was already being loaded in since it worked in my setup and in CI.

I'm guessing we just need an appropriately placed `find_package(Python REQUIRED)`?

Alternatively -- is there a more lightweight way of serializing that bytecode into a cpp?

**Comment by lgritz** [comment]
@tgrant-nv were you building with USE_PYTHON=OFF?

Python is needed for the python bindings to the oslquery library, so there is a find_package(Python) already happening at some point. And also python must be available in order to run the testsuite (which are all driving by python scripts).

**Comment by fpsunflower** [comment]
Makes sense. For windows, it seems that using `Python_EXECUTABLE` is preferable, at least in my setup the one in the path that run when you just use `python` is the wrong one, while cmake finds the right one via the `pybind` dependency.

I have yet to get the testuite fully working on windows - but I'm wondering if we should use the same strategy there ... (since I don't believe the `#!` will be picked up).
